app.ts

```diff
@@ -2,36 +2,52 @@ import express, { Express } from "express";
 import session from "express-session";
 import { PrismaSessionStore } from "@quixo3/prisma-session-store";
 import dotenv from "dotenv";
+import cors from "cors";
+import rateLimit from "express-rate-limit";
 
 dotenv.config();
 
 const app = (store: PrismaSessionStore): Express => {
-	const server: Express = express();
-
-	// Middleware
-	server.use(
-		session({
-			cookie: {
-				maxAge: 7 * 24 * 60 * 60 * 1000, // 1 week in milliseconds
-			},
-			secret: process.env.SESSION_SECRET || "",
-			resave: true,
-			saveUninitialized: true,
-			store: store,
-		})
-	);
-	server.use(express.json());
-	server.use(express.urlencoded({ extended: true }));
-
-	// Route handlers
-
-	// This is a test route that will be removed in the future
-	// It is here to test that the testing pipeline works
-	server.get("/", (req, res) => {
-		res.status(200).send("Hello World!");
-	});
-
-	return server;
+    const server: Express = express();
+
+    // Middleware
+    server.use(
+        session({
+            cookie: {
+                maxAge: 7 * 24 * 60 * 60 * 1000, // 1 week in milliseconds
+            },
+            secret: process.env.SESSION_SECRET || "",
+            resave: true,
+            saveUninitialized: true,
+            store: store,
+        })
+    );
+    server.use(express.json());
+    server.use(express.urlencoded({ extended: true }));
+
+    // Enable CORS app wide, for more information on how to use this package
+    // please refer to https://www.npmjs.com/package/cors
+    server.use(cors());
+
+    // Enable express-rate-limit, for more information on how to use this package
+    // please refer to https://www.npmjs.com/package/express-rate-limit
+    const limiter = rateLimit({
+        windowMs: 1000, // 1000 milliseconds / 1 second
+        max: 2, // Limit each IP to 2 requests per `window` (here, per 1 second)
+        standardHeaders: true, // Return rate limit info in the `RateLimit-*` headers
+        legacyHeaders: false, // Disable the `X-RateLimit-*` headers
+    });
+    server.use(limiter);
+
```

Comment:
The one thing I would suggest is to add the CORS and rate limit middleware before the session middleware. This way if a request is blocked by either CORS or rate limiting, the session middleware won't run --> this reduces (very slightly) server computations and DB queries. More efficient and reduces server load if there are a lot of requests.

Reply:
Good idea! Now that you point it out it makes a lot more sense then leaving it like this, i'll readjust this in the next commit. Thanks for your attention to detail.